test/topology_experimental_raft/test_tombstone_gc.py

```diff
@@ -198,3 +198,38 @@ async def verify_tombstone_gc(tombstone_mark, timeout=30):
 
             # test #3: the tombstones are cleaned up after the node is started again
             await verify_tombstone_gc(tombstone_mark)
+
+
+@pytest.mark.asyncio
+async def test_group0_state_id_failure(manager: ManagerClient):
+    """
+    Issue #21117 regression test.
+
+    Make sure that the "endpoint_state_map does not contain endpoint" warning is not triggered when the state_id is updated.
+
+    Arrange:
+        - start 1 node
+    Act:
+        - restart the node
+    Assert:
+        - the "endpoint_state_map does not contain endpoint" didn't appear in the log
+    """
+
+    # Start the node
+
+    srv = await manager.server_add()
+
+    cql = manager.get_cql()
+    await wait_for_cql_and_get_hosts(cql, [srv], time.time() + 60)
```

Comment:
BTW waiting for cql is probably not necessary since you're not doing any CQL ops.

Reply:
right, I though it is needed to make sure the node is fully up, but it looks like it is not needed (the test works also without these and actually runs faster then)

Reply:
`wait_for_cql_and_get_hosts` is not waiting for Scylla, it's waiting for the driver; since there's no driver usage in this test, it's unnecessary